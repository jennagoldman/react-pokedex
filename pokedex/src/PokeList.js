import React, { Component } from 'react';
import pokemonData from './data';
import PokeItem from './PokeItem';

export default class PokeList extends Component {
    render() {
        return (
            <main>
                <section className="search-container">

                </section>
                <section className="pokemon-list">
                    <ul>
                        { pokemonData.map(pokemon => <PokeItem pokemon={pokemon} key={pokemon.species_id} />)}
                    </ul>
                </section>
            </main>
        )
    }
}