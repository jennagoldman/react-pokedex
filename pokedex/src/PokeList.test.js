import React from 'react';
import PokeList from './PokeList';
import pokemonData from './data'
import renderer from 'react-test-renderer';

it('App renders correctly', () => {
  const tree = renderer
    .create(<PokeList data={pokemonData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});