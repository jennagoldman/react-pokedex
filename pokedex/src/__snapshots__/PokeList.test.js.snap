// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<main>
  <section
    className="search-container"
  />
  <section
    className="pokemon-list"
  >
    <ul>
      <li
        className="pokemon-item"
      >
        <h2>
          venusaur
        </h2>
        <img
          alt="venusaur"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
        />
        <p>
          HP: 
          80
        </p>
        <p>
          Types:
          <span>
             
            grass
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          bulbasaur
        </h2>
        <img
          alt="bulbasaur"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
        />
        <p>
          HP: 
          45
        </p>
        <p>
          Types:
          <span>
             
            grass
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          venusaur-mega
        </h2>
        <img
          alt="venusaur-mega"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003_f2.png"
        />
        <p>
          HP: 
          80
        </p>
        <p>
          Types:
          <span>
             
            grass
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          ivysaur
        </h2>
        <img
          alt="ivysaur"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/002.png"
        />
        <p>
          HP: 
          60
        </p>
        <p>
          Types:
          <span>
             
            grass
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          charmander
        </h2>
        <img
          alt="charmander"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/004.png"
        />
        <p>
          HP: 
          39
        </p>
        <p>
          Types:
          <span>
             
            fire
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          charmeleon
        </h2>
        <img
          alt="charmeleon"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/005.png"
        />
        <p>
          HP: 
          58
        </p>
        <p>
          Types:
          <span>
             
            fire
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          charizard
        </h2>
        <img
          alt="charizard"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/006.png"
        />
        <p>
          HP: 
          78
        </p>
        <p>
          Types:
          <span>
             
            fire
          </span>
          <span>
            , 
            flying
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          charizard-mega-x
        </h2>
        <img
          alt="charizard-mega-x"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/006_f2.png"
        />
        <p>
          HP: 
          78
        </p>
        <p>
          Types:
          <span>
             
            fire
          </span>
          <span>
            , 
            dragon
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          charizard-mega-y
        </h2>
        <img
          alt="charizard-mega-y"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/006_f3.png"
        />
        <p>
          HP: 
          78
        </p>
        <p>
          Types:
          <span>
             
            fire
          </span>
          <span>
            , 
            flying
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          squirtle
        </h2>
        <img
          alt="squirtle"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/007.png"
        />
        <p>
          HP: 
          44
        </p>
        <p>
          Types:
          <span>
             
            water
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          wartortle
        </h2>
        <img
          alt="wartortle"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/008.png"
        />
        <p>
          HP: 
          59
        </p>
        <p>
          Types:
          <span>
             
            water
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          blastoise
        </h2>
        <img
          alt="blastoise"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/009.png"
        />
        <p>
          HP: 
          79
        </p>
        <p>
          Types:
          <span>
             
            water
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          blastoise-mega
        </h2>
        <img
          alt="blastoise-mega"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/009_f2.png"
        />
        <p>
          HP: 
          79
        </p>
        <p>
          Types:
          <span>
             
            water
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          caterpie
        </h2>
        <img
          alt="caterpie"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/010.png"
        />
        <p>
          HP: 
          45
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          metapod
        </h2>
        <img
          alt="metapod"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/011.png"
        />
        <p>
          HP: 
          50
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            NA
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          beedrill
        </h2>
        <img
          alt="beedrill"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/015.png"
        />
        <p>
          HP: 
          65
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          weedle
        </h2>
        <img
          alt="weedle"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/013.png"
        />
        <p>
          HP: 
          40
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          kakuna
        </h2>
        <img
          alt="kakuna"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/014.png"
        />
        <p>
          HP: 
          45
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          beedrill-mega
        </h2>
        <img
          alt="beedrill-mega"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/015_f2.png"
        />
        <p>
          HP: 
          65
        </p>
        <p>
          Types:
          <span>
             
            bug
          </span>
          <span>
            , 
            poison
             
          </span>
        </p>
      </li>
      <li
        className="pokemon-item"
      >
        <h2>
          pidgey
        </h2>
        <img
          alt="pidgey"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/016.png"
        />
        <p>
          HP: 
          40
        </p>
        <p>
          Types:
          <span>
             
            normal
          </span>
          <span>
            , 
            flying
             
          </span>
        </p>
      </li>
    </ul>
  </section>
</main>
`;
